File: frontend/src/types/candidate.ts
export interface LeadCandidate {
  id: string
  source: string
  source_ref: string | null
  company_name: string
  website: string | null
  category: string | null
  llm_summary: string | null
  llm_fit_score: number | null
  status: 'pending' | 'promoted' | 'dismissed'
  discovered_at: number
  reviewed_at: number | null
  promoted_lead_id: string | null
}

export interface CandidatesPage {
  total: number
  page: number
  page_size: number
  items: LeadCandidate[]
}

export interface HarvestRequest {
  query: string
  max_results?: number
}

export interface HarvestResponse {
  discovered: number
  skipped: number
}

File: backend/app/schemas/candidate.py
from pydantic import BaseModel


class LeadCandidateRead(BaseModel):
    id: str
    source: str
    source_ref: str | None
    company_name: str
    website: str | None
    category: str | None
    llm_summary: str | None
    llm_fit_score: int | None
    status: str
    discovered_at: int
    reviewed_at: int | None
    promoted_lead_id: str | None

    model_config = {"from_attributes": True}


class LeadCandidateCreate(BaseModel):
    source: str = "manual"
    source_ref: str | None = None
    company_name: str
    website: str | None = None
    category: str | None = None
    llm_summary: str | None = None
    llm_fit_score: int | None = None


class HarvestRequest(BaseModel):
    query: str
    max_results: int = 25


class HarvestResponse(BaseModel):
    discovered: int
    skipped: int
